Reworks transformation sting to be temporarily in living mobs, forever in dead mobs

MrMelbert opened the pull request:
## About The Pull Request

- Reworks transformation sting. 
   - Transformation sting is now temporary, lasting 8 minutes (number not final) in humans. 
      - If used on a monkey, it lasts forever instead. 
      - While the target mob is dead or in stasis, the duration will not progress, making it functionally infinite until revived and taken off stasis, where it will resume its timer where it left off.
   - Chemical cost reduced to 33
   - DNA cost reduced to 2

- Removes TRAIT_NO_TRANSFORMATION_STING, instead just checks for TRAIT_NO_DNA_COPY
   - These were essentially the same traits, so I just combined the two

- Organizes some trait lists alphabetically

- Adds TRAIT_STASIS, to allow for reacting to mobs entering and exiting stasis via COMSIGS
   - Everything that checks IS_IN_STASIS now checks HAS_TRAIT TRAIT_STASIS, which is probably more performant, so that's a bonus. 

## Why It's Good For The Game

A lot of people don't like the current iteration of Transformation Sting, me included

Right now it's only use is for a meme - you make the entire station into felinids until you get lynched, and that's it. 

It's not really a healthy ability for ling's current kit, so this pr attempts to soft rework it to make it a bit more in line with how ling should be acting - turning it into a source of short term confusion between people, or using it on a body to cover your tracks. 

This accomplish it two fold - One, it is now cheap enough to use twice in rapid succession, allowing for quick on-the-spot "BE CONFUSED" situations while you abscond. Two, as mentioned in the last paragraph, you can poke a body of someone you murder to obfuscate the crime scene and maybe help out in taking over someone's identity. 

## Changelog

:cl: Melbert
balance: Transformation sting now lasts 8 minutes, down from permanent. However, the effect is paused for dead and stasis mobs, making it permanent SO LONG AS they stay dead or in stasis. The effect is also permanent if used on a monkey. 
balance: Transformation sting now costs 33 chemicals, down from 50. 
balance: Transformation sting now costs 2 dna points, down from 3. 
fix: Transformation sting works on monkeys again.
refactor: Refactored a bit of human randomization. 
/:cl:

MrMelbert committed:
- Reworks transformation sting to be temporarily while not dead
- Removes TRAIT_NO_TRANSFORMATION_STING
- Alphabetizes a few trait lists
- Adds TRAIT_STASIS, changes IS_IN_STASIS to be a trait check

MrMelbert committed:
Fixes

ATH1909 commented:
literally george orwell's novel, 1984

MrMelbert committed:
Adds a unit test

MrMelbert commented:
Drafted cause I'm making a screenshot test for it and it's doing some goofy things
![image](https://github.com/tgstation/tgstation/assets/51863163/b4bb3bff-2423-47e1-ba82-f712f42657f7)

Serxule commented:
i mean i did dislike the gimmick of turning everyone on the station into 1 person but also i feel like it being temporary is kinda useless other than turning someone elses corpse into the known ling and changing yourself so people think ling dead etc

VladinXXV commented:
you stole this idea from that ling round I stung a heretic you killed to get them revived, where's my cut?

Striders13 commented:
It should just be permanent when used on a corpse imo. You already got killed, I don't think you can complain about getting trolled by ling.

HoodMilk commented:
Transformation sting should honestly just be axed but this is a good start.

DaydreamIQ commented:
This is a great way to force people to actually be interesting with it instead of forcing everyone to spend the round as the wrong person because there's no way to reverse it without backing up your dna beforehand

MrMelbert committed:
Reorganizing test

necromanceranne commented:
Transformation sting was _meant_ to hide corpses, but literally only ever gets used to meme people.

One of my formative rounds was some clown turning people into Ronald McDonald and honestly, that was very fucking funny. But nobody has yet to top such a round because they really just use it to turn everyone into someone else's static/their own static.

MrMelbert committed:
Fixes some confusing aspects of DNA

MrMelbert committed:
A slew of DNA related fixes:
- Fixes humans randomizing multiple times
- Fixes some monkey tail shenanigans
- Refactors how random features is done
This was done to fix a lot of UI / UE / UF related bugs coming out of unsynced / inaccurate DNA

MrMelbert committed:
Comment

MrMelbert committed:
Merge branch 'master' into transsting_two

MrMelbert committed:
Misplaced trait

MrMelbert committed:
Merge branch 'master' into transsting_two

MrMelbert committed:
Testing

MrMelbert committed:
Screenshot test fix

MrMelbert committed:
Monkey tail fix?

MrMelbert committed:
Monkey test fix

LemonInTheDark commented on code/modules/unit_tests/changeling.dm:
```diff
@@ -0,0 +1,95 @@
+	var/ling_name = "Is-A-Changeling"
+	var/base_victim_name
+	var/last_frame = 1
+	var/icon/final_icon
```
user above me doesn't know about the icon cache

MrMelbert replied:
what if you told me what it is instead of telling me i don't know what it is 👉 👈

LemonInTheDark replied:
I thought jolly's pr (https://github.com/tgstation/tgstation/pull/78799) would get linked but it didn't so I'll just info dump here
Icons are basically wrappers around file references, they become modifiable while they are "cached", or loaded in.

The problem with this is byond was made by memory conscious devs, so an icon doesn't STAY cached forever.
There's a first in first out queue sitting in memory somewhere with a set length, and after X amount of icons are loaded, the oldest one gets dropped.

This'd ALSO be fine if you could treat dropped icons like normal ones, but you can't. Trying to modify them, like with the Insert() proc, will instead throw an error. Just says "bad icon" and goes about its day.

It's good to account for this when holding refs to icons, since it can cause some very confusing errors

The only way to avoid bad icon errors is to reload the icon, basically by doing something like this 
`final_icon = icon(final_icon)`

This reinserts it into the cache, and makes it safe to modify again.

LemonInTheDark replied:
The real headache is this is documented fucking nowhere, not even in my writing, and hardly anyone (me and ninja I think) know about it.
Worst fucking runtime name in the history of the universe I swear man.

SinfulBliss commented:
A better way to prevent the “turning everyone to felinids” meme is to just limit the amount of people it could be used on at a time, 8 minutes is too short to be able to effectively rely on this serving as identity subterfuge, which was the whole use behind it previously.

I think antag mechanics shouldn’t be getting nerfed around people salting that their static OCs are lost for a round when they get stung.
